### PR TITLE
[AC-1917-without-domain-changes] All other form updates other than domains

### PR DESCRIPTION
**Pull request by shrirupa**
This is split branch from AC-1917 and has all other changes other than domain

### What Changed

**On Ab testing page**
- [ ] EditMode.js - Reschedule and Finalize test button should be a submit button inside a `<form>`

**On Report Builder**
- [ ] MetricsDrawer - this should be wrapped in a `<form>`

**On Snippet**
- [ ] CreatePage and EditPage Save Snippet and Create Snippet should have associated form id and type submit.

**On JoinForm**
- [ ] JoinForm submits on form submit instead of button click.

### How To Test

- onSubmit of these newly added forms work correctly

### To Do

- [ ] Address feedback
- [ ] Check if all form have submit buttons.

### CODE REVIEW

File: src/pages/abTesting/EditMode.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { connect } from 'react-redux';
 import { reduxForm, getFormValues } from 'redux-form';
```
Comment (nstanard) [inline src/pages/abTesting/EditMode.js:4]: 🟢  - submit handler event fires, lgtm
image

File: src/pages/join/components/JoinForm.js [hunk 2]
```diff
@@ -54,7 +54,10 @@ export class JoinForm extends Component {
     this.setState({ recaptcha_response: response }, this.props.handleSubmit(this.onSubmit));
```
Comment (nstanard) [inline src/pages/join/components/JoinForm.js:54]: 🟢  - submit handler event fires, lgtm
image

Comment (nstanard) [reply src/pages/join/components/JoinForm.js:54]: Although... for some reason I can't actually create an account right now. I don't think it's related to this PR but i'll double check this tomorrow morning.

File: src/pages/join/components/JoinForm.js [hunk 3]
```diff
@@ -54,7 +54,10 @@ export class JoinForm extends Component {
...
   };
 
-  executeRecaptcha = () => this.state.reCaptchaInstance.execute();
+  executeRecaptcha = event => {
+    event.preventDefault();
+    this.state.reCaptchaInstance.execute();
```
Comment (nstanard) [inline src/pages/join/components/JoinForm.js:59]: You might want to add a `return` so this method returns `this.state.reCaptchaInstance.execute();` like the arrow functionw as doing already. (lowers risk of unintended consequences)

File: src/pages/snippets/EditPage.js [hunk 4]
```diff
@@ -95,18 +95,18 @@ export default class EditPage extends React.Component {
     }
 
     return (
-      <Page
```
Comment (nstanard) [inline src/pages/snippets/EditPage.js:98]: 🟢  - submit handler event fires, lgtm (for create and edit)

image

File: src/pages/reportBuilder/components/MetricsForm/MetricsForm.js [hunk 5]
```diff
@@ -97,7 +99,7 @@ export default function MetricsForm(props) {
   //Needs this for current tests as hibana is not enabled for tests but is required for the Drawer component
   const { DrawerFooter = Drawer.Footer } = props;
   return (
-    <>
+    <Form id="reportbuilder-metrics-form" onSubmit={handleApply}>
```
Comment (nstanard) [inline src/pages/reportBuilder/components/MetricsForm/MetricsForm.js:102]: 🟢  - submit handler event fires, lgtm

image

File: src/pages/abTesting/EditMode.js [hunk 6]
```diff
@@ -77,29 +77,43 @@ export class EditMode extends Component {
...
 
     return {
       content: 'Update Test',
-      onClick: handleSubmit(this.handleUpdateScheduled),
+      type: 'submit',
     };
   };
 
+  getSubmitAction = () => {
+    const { test, rescheduling } = this.props;
```
Comment (nstanard) [inline src/pages/abTesting/EditMode.js:104]: note to self: Technically we want to test/verify all three action types when we move this along

File: src/pages/join/components/JoinForm.js (hunk 3 above)
Comment (nstanard) [reply src/pages/join/components/JoinForm.js:59]: 🤔  ... now I'm not sure sure actually. Probably doesn't make a difference. But the arrow function did return the execute method call.

File: src/pages/join/components/JoinForm.js (hunk 2 above)
Comment (nstanard) [reply src/pages/join/components/JoinForm.js:54]: Whew, finally got the create to go through. (it was just sitting there loading the first couple times I tried .. probably VPN)
image

**Review by nstanard — APPROVED** [review]
One small note/comment but this LGTM!

File: src/pages/join/components/JoinForm.js (hunk 3 above)
Comment (shrirupa) [reply src/pages/join/components/JoinForm.js:59]: Good catch!

**Pull request merged**